src/kili/use_cases/utils.py

```diff
@@ -0,0 +1,89 @@
+"""Utils for use cases."""
+from itertools import chain
+from typing import Dict, List, Optional, cast
+
+from kili.adapters.kili_api_gateway import KiliAPIGateway
+from kili.adapters.kili_api_gateway.helpers.queries import QueryOptions
+from kili.core.utils import pagination
+from kili.domain.asset import AssetExternalId, AssetFilters, AssetId
+from kili.domain.asset.helpers import check_asset_identifier_arguments
+from kili.domain.project import ProjectId
+from kili.domain.types import ListOrTuple
+from kili.exceptions import NotFound
+
+
+class UseCasesUtils:
+    """Utils for use cases."""
+
+    def __init__(self, kili_api_gateway: KiliAPIGateway) -> None:
+        """Init."""
+        self.kili_api_gateway = kili_api_gateway
+
+    def get_asset_ids_or_throw_error(
+        self,
+        asset_ids: Optional[ListOrTuple[AssetId]],
+        external_ids: Optional[ListOrTuple[AssetExternalId]],
+        project_id: Optional[ProjectId],
+    ) -> List[AssetId]:
+        """Check if external id to internal id conversion is valid and needed."""
+        check_asset_identifier_arguments(project_id, asset_ids, external_ids)
+
+        if asset_ids is None and external_ids is not None and project_id is not None:
+            id_map = self.infer_ids_from_external_ids(external_ids, project_id)
+            resolved_asset_ids = [id_map[ext_id] for ext_id in external_ids]
+        else:
+            resolved_asset_ids = asset_ids or []
+
+        return cast(List[AssetId], resolved_asset_ids)
```

Comment:
the cast at the end should not be needed (and is actually wrong)

if asset_ids input parameter is a Tuple of AssetIds, then resolved_asset_ids will be a Tuple, the return type too

so the return type can be a ListOrTuple too